discord/ext/commands/bot.py

```diff
@@ -568,20 +568,31 @@ def _call_module_finalizers(self, lib, key):
                 if _is_submodule(name, module):
                     del sys.modules[module]
 
-    def _load_from_module_spec(self, lib, key):
+    def _load_from_module_spec(self, spec, key):
         # precondition: key not in self.__extensions
+        def unload(e):
+            self._remove_module_references(lib.__name__)
+            self._call_module_finalizers(lib, key)
+            raise errors.ExtensionFailed(key, e) from e
+
+        lib = importlib.util.module_from_spec(spec)
+        try:
+            spec.loader.exec_module(lib)
+        except Exception as e:
+            unload(e)
```

Comment:
What is this even doing here?

If the import failed then the `lib` reference is probably broken.